### ISSUE TITLE
BigQuery Transactions with CASE statements are unparseable

### ISSUE THREAD

**Issue by nathadfield** [sqlfluff/sqlfluff#5461]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running SQLFluff on a BigQuery [multi-statement transaction](https://cloud.google.com/bigquery/docs/transactions) with a column that uses a CASE statement, an error relating to an `unparseable section` is raised.

```
L:   4 | P:   3 |  PRS | Line 4, Position 3: Found unparsable section: 'INSERT
                       | INTO `my_project.my_dataset.my_ta...'
L:   8 | P:  47 | LT02 | Expected line break and no indent before 'END'.
                       | [layout.indent]
L:   8 | P:  52 |  PRS | Line 8, Position 52: Found unparsable section: 'AS
                       | col3,\n  FROM `my_project.my_dataset.m...'
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
```

### Expected Behaviour

The expected behaviour here is that no errors should be raised. 

### Observed Behaviour

I believe the problem is being caused by the `END` that concludes a case statement being confused by the `END` that should exist at the end of specifying a transaction. 

### How to reproduce

Here is the SQL that you can use to replicate that problem.

```
BEGIN
  BEGIN TRANSACTION;

  INSERT INTO `my_project.my_dataset.my_table`
  SELECT
    col1,
    col2,
    CASE WHEN col1 > col2 THEN False ELSE True END AS col3,
  FROM `my_project.my_dataset.my_other_table`
  ;

  COMMIT TRANSACTION
  ;
END
;
```

Run this with the following command
```
sqlfluff lint test.sql --dialect bigquery
```

### Dialect

BigQuery

### Version

```
$ sqlfluff --version                                                              
sqlfluff, version 2.3.5
```

### Configuration

```
[tool.sqlfluff.core]
exclude_rules = "L016,L031,L060,L034,ST07"
large_file_skip_byte_limit = 0
[tool.sqlfluff.indentation]
tab_space_size = 2
[tool.sqlfluff.rules.convention.terminator]
multiline_newline = true
require_final_semicolon = true
[tool.sqlfluff.rules.convention.select_trailing_comma]
select_clause_trailing_comma = "require"
[tool.sqlfluff.rules.convention.quoted_literals]
preferred_quoted_literal_style = "double_quotes"
[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"
[tool.sqlfluff.rules.capitalisation.functions]
extended_capitalisation_policy = "upper"
[tool.sqlfluff.rules.capitalisation.types]
extended_capitalisation_policy = "upper"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by salma-ol** [comment]
I am facing the same issue with SQL code (BigQuery dialect) that contains a CASE inside a transaction. This issue seems to have started with the 2.3.5 release as the code is parsable in 2.3.4.
